test/functional/tests/io_class/test_io_class_file.py

```diff
@@ -233,30 +381,35 @@ def test_ioclass_file_extension_preexisting_filesystem():
             ioclass_config_path=ioclass_config_path,
         )
 
-    with TestRun.step(f"Add device with preexisting data as a core."):
-        core = casadm.add_core(cache, core_dev=core.core_device)
+    with TestRun.step("Add device with preexisting data as a core"):
+        core = casadm.add_core(cache, core_dev=core_device)
 
-    with TestRun.step("Load IO class config."):
+    with TestRun.step("Load IO class config"):
         casadm.load_io_classes(cache_id=cache.cache_id, file=ioclass_config_path)
 
-    with TestRun.step("Mount core and flush cache."):
+    with TestRun.step("Mount core and flush cache"):
         core.mount(mountpoint)
         cache.flush_cache()
 
-    with TestRun.step(f"Write to file with cached extension and check if they are cached."):
-        for ext in extensions:
-            (
+    with TestRun.step("Write to file with cached extension and check if they are cached"):
+        for extension in extensions:
+            dd = (
                 Dd()
                 .input("/dev/zero")
-                .output(f"{mountpoint}/test_file.{ext}")
+                .output(f"{mountpoint}/test_file.{extension}")
                 .count(dd_count)
                 .block_size(dd_size)
-                .run()
             )
+            dd.run()
             sync()
-            dirty = cache.get_io_class_statistics(io_class_id=ioclass_id).usage_stats.dirty
-            if dirty.get_value(Unit.Blocks4096) != (extensions.index(ext) + 1) * dd_count:
-                TestRun.LOGGER.error(f"Wrong amount of dirty data ({dirty}).")
+
+            io_class_statistics = cache.get_io_class_statistics(io_class_id=ioclass_id)
+            dirty_stat = io_class_statistics.usage_stats.dirty
+            if (
+                dirty_stat.get_value(Unit.Blocks4096)
+                != (extensions.index(extension) + 1) * dd_count
```

Comment:
use 'range exceptions' in line 395, and replace this index resolution with simple variable